Lib/test/test_syslog.py

```diff
@@ -78,6 +79,52 @@ def logger():
         finally:
             sys.setswitchinterval(orig_si)
 
+    def test_subinterpreter_syslog(self):
+        code = dedent('''
+            import syslog
+            catch_error = False
+            try:
+                syslog.syslog('foo')
+            except RuntimeError:
+                catch_error = True
+
+            assert(catch_error == True)
+        ''')
+        res = support.run_in_subinterp(code)
+        self.assertEqual(res, 0)
```

Comment:
It would be helpful to add a comment that `syslog.syslog()` is only disallowed if `syslog.openlog()` (or `syslog.syslog()`) hasn't been called yet in the main interpreter.

In fact, it might also be worth testing that second case:

```suggestion
    def test_subinterpreter_syslog(self):
        # syslog.syslog() is not allowed in subinterpreters, but only if
        # syslog.openlog() hasn't been called in the main interpreter yet.
        with self.subTest('before openlog()'):
            code = dedent('''
                import syslog
                catch_error = False
                try:
                    syslog.syslog('foo')
                except RuntimeError:
                    catch_error = True

                assert(catch_error is True)
            ''')
            res = support.run_in_subinterp(code)
            self.assertEqual(res, 0)

        syslog.openlog()
        with self.subTest('after openlog()'):
            code = dedent('''
                import syslog
                syslog.syslog('foo')
            ''')
            res = support.run_in_subinterp(code)
            self.assertEqual(res, 0)
```